include/drivers/i2c.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#ifndef DRIVERS_I2C_H
+#define DRIVERS_I2C_H
+
+#include <dm.h>
+#include <error.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#define I2C_OPS(dev) ((struct i2c_driver_ops *)((dev)->drv->ops))
+
+struct i2c_driver_ops {
+	int (*read)(struct device *dev, uint8_t *data);
+	int (*start)(struct device *dev, int i2c_op);
+	int (*stop)(struct device *dev);
+	int (*write)(struct device *dev, uint8_t data);
+};
+
+enum I2COps {
+	I2C_READ,
+	I2C_WRITE,
+};
+
+/**
+ * TODO(@drewwalters96): Add function documentation.
+ */
+static inline int
+i2c_probe(struct device *dev)
```

Comment:
All of these `i2c_*` functions should be moved to `i2c.c`, since they're way too big to be inlined.